### PR TITLE
Fixing issue where AspireManifestPublishOutputPath is specified

### LINKED ISSUE #1076: GenerateAspireManifest target with value in AspireManifestPublishOutputPath errors out
When setting an `AspireManifestPublishOutputPath` property on the app host and running `dotnet msbuild /t:GenerateAspireManifest` the target fails

```
D:\AppCacheFiles\nuget\aspire.hosting\8.0.0-preview.1.23557.2\build\Aspire.Hosting.targets(118,5): error MSB3073: The c
ommand "..." exited with code -532462766. [D:\source\repos\AspireApp7\AspireApp7.AppHost\AspireApp7.AppHost.csproj]
```

### PR DESCRIPTION
**Pull request by joperezr**
Fixes https://github.com/dotnet/aspire/issues/1076

Before this change, setting AspireManifestPublishOutputPath would only work if you passed in a path with a filename, as well as ensuring the output directory existed. With this change, AspireManifestPublishOutputPath will now be pointing to the output folder where manifest.json file will be created, and it won't be required for t he folder to exist. This will allow people to set the property in their AppHost projects and work as expected.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2388)

### CODE REVIEW

**Review by baronfel — APPROVED** [review]

File: src/Aspire.Hosting/build/Aspire.Hosting.targets [hunk 1]
```diff
@@ -224,15 +224,24 @@ internal static class ]]>%(ClassName)<![CDATA[
 
   <PropertyGroup>
     <AspirePublisher Condition="'$(AspirePublisher)' == ''">manifest</AspirePublisher>
-    <AspireManifestPublishOutputPath Condition="'$(AspireManifestPublishOutputPath)' == ''">$(_AspireIntermediatePath)manifest.json</AspireManifestPublishOutputPath>
+    <AspireManifestPublishOutputPath Condition="'$(AspireManifestPublishOutputPath)' == ''">$(_AspireIntermediatePath)</AspireManifestPublishOutputPath>
   </PropertyGroup>
 
+  <Target Name="_EnsureAspireManifestPublishOutputPathExists" BeforeTargets="GenerateAspireManifest">
```
Comment (eerhardt) [inline src/Aspire.Hosting/build/Aspire.Hosting.targets:230]: You don't need both `DependsOnTargets` and `BeforeTargets`. You can just pick one. My preference would be `DependsOnTargets`.

**Comment by mitchdenny** [comment]
@joperezr I wouldn't over index on those docs. They are going to get a major refresh, the entire manifest has changed quite a bit in P4.

File: src/Aspire.Hosting/build/Aspire.Hosting.targets (hunk 1 above)
Comment (joperezr) [reply src/Aspire.Hosting/build/Aspire.Hosting.targets]: Oops, you are right, I started with BT but then switched to DOT and forgot to remove the older one. Fixed.

**Comment by joperezr** [comment]
> @joperezr I wouldn't over index on those docs. They are going to get a major refresh, the entire manifest has changed quite a bit in P4.

Sounds good, I'll log an issue in the docs repo to remember to update that piece of the docs.

**Review by eerhardt — APPROVED** [review]

**Pull request merged**